metagraph/src/annotation/taxonomy/tax_classifier.hpp

```diff
@@ -0,0 +1,191 @@
+#ifndef __TAX_CLASSIFIER_HPP__
+#define __TAX_CLASSIFIER_HPP__
+
+#ifdef TESTING
+#define PRIVATE_TESTABLE public
+#define PROTECTED_TESTABLE public
+#else
+#define PRIVATE_TESTABLE private
+#define PROTECTED_TESTABLE protected
+#endif
+
+#include <tsl/hopscotch_set.h>
+#include <tsl/hopscotch_map.h>
+
+#include "graph/annotated_dbg.hpp"
+
+namespace mtg {
+namespace annot {
+
+using TaxId = std::uint32_t;
+using ChildrenList = tsl::hopscotch_map<TaxId, std::vector<TaxId>>;
+
+class TaxonomyBase {
+public:
+    using KmerId = annot::MultiLabelEncoded<std::string>::Index;
+    using node_index = graph::SequenceGraph::node_index;
+
+    enum LabelType {
+        UNASSIGNED, GEN_BANK, KRAKEN
```

Comment:
what do these label types represent? what is a Kraken label here?

Reply:
ah, after reading above, it looks like `KRAKEN`-type labels are taxonomic IDs. So we should call this `TAXID` instead of `KRAKEN`

Reply:
Thanks for the review! This `enum` contains a list of labels formats where we need to hardcode how to get the accession version or the taxid.

For example:

- one GEN_BANK label looks like this: ">gi|1070643132|ref|**NC_031224.1**| Arthrobacter phage Mudcat, complete genome"
- one Kraken2 label looks like: ">kraken:taxid|2016032|**NC_047834.1** Alteromonas virus vB_AspP-H4/4, complete genome"

Maybe there are better ideas than hardcoding a "get_accession_version" method for all these formats, but I didn't see any other way around this